mbuild/compound.py

```diff
@@ -2235,6 +2346,208 @@ def _energy_minimize_openbabel(
+                        fixed_compounds = [fixed_compounds]
+
+            for fixed_temp in fixed_compounds:
+                if isinstance(fixed_temp, list):
+                    # if the list is length two
+                    # we will validate that 3 bool values have been passed
+                    if len(fixed_temp) == 2:
+                        if len(fixed_temp[1]) == 3:
+                            p1 = fixed_temp[0]
+                            check0 = isinstance(fixed_temp[1][0], bool)
+                            check1 = isinstance(fixed_temp[1][1], bool)
+                            check2 = isinstance(fixed_temp[1][2], bool)
+                            if check0 and check1 and check2:
+                                dims = [
+                                    fixed_temp[1][0],
+                                    fixed_temp[1][1],
+                                    fixed_temp[1][2],
+                                ]
+                            else:
+                                raise MBuildError(
+                                    f"Expected bool values for which directions are fixed."
+                                    f"Found instead {str(type(fixed_temp[1][0]))}, "
+                                    f"{str(type(fixed_temp[1][1]))}, "
+                                    f"{str(type(fixed_temp[1][2]))}."
+                                )
+                        else:
+                            raise MBuildError(
+                                "Expected tuple or list of length 3 to set"
+                                "which dimensions to fix motion, "
+                                f"{str(len(fixed_temp[1]))} found."
+                            )
+                    # if the list is defined as [[compound1],[compound2]]
+                    elif len(fixed_temp) == 1:
+                        p1 = fixed_temp[0]
+                        dims = [True, True, True]
+                else:
+                    p1 = fixed_temp
+                    dims = [True, True, True]
+
+                all_true = False
+                if dims[0] == dims[1] == dims[2] == True:
+                    all_true = True
+
+                p1_in_compound = False
+                for successor in self.successors():
+                    if id(p1) == id(successor):
+                        p1_in_compound = True
+                if id(p1) == id(self):
+                    p1_in_compound = True
+
+                if p1_in_compound == False:
+                    raise MBuildError(
+                        f"{p1} is not a member of Compound {self}."
+                    )
+
+                if len(p1.children) == 0:
+                    index = 1
+                    for particle in self.particles():
+                        if id(p1) == id(particle):
+                            pid = index
+                            if all_true == True:
+                                ob_constraints.AddAtomConstraint(pid)
```

Comment:
I don't think you want these if elif cascades. What if dims looks like: `[False, True, True]`. Then you'll only get ob_constraints for AddAtomYConstraint, and not Z which should also be added. Maybe I'm missing the reasoning for adding them one at a time though.